pkg/servicemonitor/servicemonitor.go

```diff
@@ -23,9 +23,10 @@ type ServiceMonitor struct {
 
 func NewServiceMonitor(ctx context.Context, c client.Client, enablehypershift bool) *ServiceMonitor {
 	return &ServiceMonitor{
-		Client:   c,
-		Ctx:      ctx,
-		Comparer: &util.ResourceComparer{},
+		Client:     c,
+		Ctx:        ctx,
+		Comparer:   &util.ResourceComparer{},
+		Hypershift: enablehypershift,
```

Comment:
How is the `enableHypershift` determined? This should only be true if RMO is reconciling clusterurlmonitors deployed on HCP namespaces right?